pkgs/applications/science/math/gap/default.nix

```diff
@@ -104,34 +123,25 @@ stdenv.mkDerivation rec {
     # like the defaults the Makefile, but use gap from PATH instead of the
     # one from builddir
     installCheckFlagsArray+=(
-      "TESTGAP=gap --quitonbreak -b -m 100m -o 1g -q -x 80 -r -A"
-      "TESTGAPauto=gap --quitonbreak -b -m 100m -o 1g -q -x 80 -r"
+      "TESTGAPcore=gap --quitonbreak -b -q -r"
+      "TESTGAPauto=gap --quitonbreak -b -q -r -m 100m -o 1g -x 80"
+      "TESTGAP=gap --quitonbreak -b -q -r -m 100m -o 1g -x 80 -A"
     )
   '';
 
   postBuild = ''
     pushd pkg
-    bash ../bin/BuildPackages.sh
+    # failures are ignored unless --strict is set
+    bash ../bin/BuildPackages.sh ${lib.optionalString (!keepAll) "--strict"}
     popd
   '';
 
-  installTargets = [
-    "install-libgap"
-    "install-headers"
-  ];
-
-  # full `make install` is not yet implemented, just for libgap and headers
   postInstall = ''
-    # Install config.h, which is not currently handled by `make install-headers`
-    cp gen/config.h "$out/include/gap"
-
-    mkdir -p "$out/bin" "$out/share/gap/"
-
-    echo "Copying files to target directory"
-    cp -ar . "$out/share/gap/build-dir"
+    # make install creates an empty pkg dir. make check requires a tst dir.
```

Comment:
So you install the `tst` files just so that you can run the test suite? Isn't that overkill? Or do you consider it desirable to install test suites? What others packages do that (e.g. for `git` itself, I don't think I've ever seen its test suite being installed alongside `git` itself).

Reply:
Note that instead of a wrapper script you could also add `-DSYS_DEFAULT_PATHS="prefix"` to `CFLAGS` (Fedora does it this way).

(See also https://github.com/gap-system/gap/issues/5096 for a plan to do this "automatically " in the future

Reply:
I do not remember why I decided to do things this way. If there is a way to run the tests with the installed version of gap without installing the tests, I agree that that would be better.

Reply:
It is possible to let the installed GAP run the non-installed test suite (and I agree this is sensible and desirable).

one way would be to invoke it like this (untested; replace `$srcdir` suitably): `gap --quitonbreak -c "TestDirectory( \"$srcdir/tst/testinstall\", rec(exitGAP := true) );"

Reply:
Thanks, I will try this! It will be a nice improvement.

Reply:
I finally had a chance to test this myself and unfortunately it doesn't work that way, because some tests want to access additional files in the `tst` directory and for this assume it is available in a GAP root... but you could still do this then:
```
$prefix/bin/gap --quitonbreak -l ";$srcdir" $srcdir/tst/testinstall.g
```
The `-l ";$srcdir"` option appends `$srcdir` to the list of GAP root paths, so that code trying things like `ReadGapRoot("tst/testinstall/MatrixObj/testmatobj.g")` still works

Reply:
Thanks for testing and sorry for not doing it earlier. I appreciate the suggestion but I worry that `-l ";$srcdir"` might be equivalent to testing the build-dir version of GAP, in the sense that a GAP test run might succeed by finding things in `$srcdir` (other than `tst/`) that it wouldn't on a normal test run.

I also want to keep the package as lean as possible, but the default GAP 4.12.1 install takes 759MB, and just 6MB of those come from the `tst` directory. The current version of the package is heavier because it ships the full build dir (802MB), including the `tst` subdirectory. Therefore, I am inclined to postpone this improvement to have more time to experiment, leaving a TODO pointing to this discussion, mostly because I will still be pretty busy until next week. If further experimentation leads to the conclusion that testing the build-dir version of GAP is fine (which is likely!), I can just move the testrun from `installCheckPhase` to `checkPhase`, not ship the `tst` directory and avoid the problem altogether.

The `-DSYS_DEFAULT_PATHS` tip is interesting and I appreciate the information! Wrappers are very common in Nix land, though, and I want to investigate if this would break a [content-addressed build](https://www.tweag.io/blog/2020-11-18-nix-cas-self-references/) of GAP. Therefore, I probably will investigate both improvements at once in the near future.

I hope postponing these improvements doesn't give the wrong impression. I really appreciate the feedback and will act  on the feedback in the near future.

Reply:
Sure, it's not urgent.

BTW regarding space, you could `gzip` some data files esp. in large packages, too reduce this a bit. E.g. for the [termux package](https://github.com/termux/termux-packages/pull/12555) I suggested something like this to them:
```
	# To get them to be small enough, could gzip them in place
 	# (GAP transparently allows read access to those)
 	gzip -n pkg/primgrp/data/*.g
 	gzip -n pkg/transgrp/data/*.grp
 	gzip -n -f pkg/smallgrp/id*/*.*
 	gzip -n -f pkg/smallgrp/small*/*.*
```
This was only for the required packages primgrp, transgrp, smallgrp; it could be applied to more (e.g. `irredsol/data/*.grp`, `ctbllib/data/*.tbl`, `sglppow/lib/3hoch8/rank*`). I will also look into modifying at least those three packages to ship with pre-compressed files. (I've also used `-9` / `--best` in the past but mostly the savings don't seem to justify the slow performance. YMMV)